datasets/pl/finanse_sanctions/crawler.py

```diff
@@ -16,88 +16,107 @@
+
+from rigour.mime.types import XLSX
 from zavod import Context
 from zavod import helpers as h
 from zavod.shed.un_sc import get_legal_entities, get_persons, Regime, load_un_sc
 
-PDF_URL = "https://www.gov.pl/attachment/2fc03b3b-a5f6-4d08-80d1-728cdb71d2d6"
 POLAND_PROGRAM = "art. 118 ustawy z dnia 1 marca 2018 r. o przeciwdziałaniu praniu pieniędzy i finansowaniu terroryzmu"
 UN_SC_PREFIXES = [Regime.TALIBAN, Regime.DAESH_AL_QAIDA]
 
-KNOWN_HASHES = {
-    "https://www.gov.pl/attachment/2fc03b3b-a5f6-4d08-80d1-728cdb71d2d6": "94c0607177fec8a07ca3e7d82c3d61be36ea20ee",
-    "https://www.gov.pl/attachment/56238b34-8a26-4431-a05a-e1d039f0defa": "3b8c0419879991e4dfd663aeed7b2df3c7472c55",
-}
+PSEUDONYM_SPLITS = ["a) ", "b) ", "c) "]
 
 
 def crawl_row(context: Context, row: Dict[str, str]):
     entity = context.make("Person")
-    name = row.pop("Imiona i nazwisk")
-    birthplace = row.pop("Miejsce urodzenia")
+    name = row.pop("imiona_i_nazwiska")
+    birthplace = row.pop("miejsce_urodzenia")
     entity.id = context.make_id(birthplace, name)
     entity.add("name", name)
-    entity.add("alias", row.pop("Pseudonim").split("\n"))
+    entity.add("alias", h.multi_split(row.pop("pseudonim"), PSEUDONYM_SPLITS))
     birth_country = birthplace.split(",")[-1]
     entity.add("birthPlace", birthplace, lang="pol")
     entity.add("birthCountry", birth_country, lang="pol")
-    h.apply_date(entity, "birthDate", row.pop("Data urodzenia"))
-    entity.add("address", row.pop("location full") or None)
-    entity.add("country", row.pop("location country") or None)
+    h.apply_date(entity, "birthDate", row.pop("data_urodzenia"))
 
-    entity.add("nationality", row.pop("narodowość"))
-    entity.add("topics", "sanction")
+    other_information = row.pop("inne_informacje")
+    res = context.lookup("other_information", other_information)
+    if not res:
+        context.log.warning("No extra information lookup found", row=row)
+    else:
+        entity.add(
+            "nationality",
+            res.properties.get("nationality"),
+            lang="pol",
+            original_value=other_information,
+        )
+        entity.add(
+            "country",
+            res.properties.get("country"),
+            lang="pol",
+            original_value=other_information,
+        )
+        entity.add(
+            "address",
+            res.properties.get("address"),
+            lang="pol",
+            original_value=other_information,
+        )
 
+    entity.add("topics", "sanction")
     sanction = h.make_sanction(context, entity)
-    h.apply_date(sanction, "listingDate", row.pop("Data umieszczenia na liście"))
+    h.apply_date(sanction, "listingDate", row.pop("data_umieszczenia_na_liscie"))
     sanction.add(
-        "reason", collapse_spaces(row.pop("Uzasadnienie wpisu na listę")), lang="pol"
+        "reason",
+        collapse_spaces(row.pop("uzasadnienie_wpisu_na_liste_url")),
+        lang="pol",
     )
```

Comment:
ah sorry, the reason text is still nice in the reason prop. The link can go in sanction:sourceUrl

make_sanction automatically adds the publisher URL, and it's quite common to additionally add reference material linked in the data.